e2e/vmss.go

```diff
@@ -0,0 +1,160 @@
+
+	// Private Key generation
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create rsa private key: %q", err)
+	}
+
+	// Validate Private Key
+	err = privateKey.Validate()
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate private key: %q", err)
+	}
+
+	publicRsaKey, err := ssh.NewPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert private to public key: %q", err)
+	}
+
+	pubKeyBytes := ssh.MarshalAuthorizedKey(publicRsaKey)
+
+	// Get ASN.1 DER format
+	privDER := x509.MarshalPKCS1PrivateKey(privateKey)
+
+	// pem.Block
+	privBlock := pem.Block{
+		Type:    "RSA PRIVATE KEY",
+		Headers: nil,
+		Bytes:   privDER,
+	}
+
+	// Private key in PEM format
+	privatePEM := pem.EncodeToMemory(&privBlock)
+
+	pollerResp, err := cloud.vmssClient.BeginCreateOrUpdate(
+		ctx,
+		"MC_agentbaker-e2e-tests_agentbaker-e2e-test-cluster_eastus",
+		name,
+		armcompute.VirtualMachineScaleSet{
+			Location: to.Ptr(location),
+			SKU: &armcompute.SKU{
+				Name:     to.Ptr("Standard_DS2_v2"),
+				Capacity: to.Ptr[int64](1),
+			},
+			Properties: &armcompute.VirtualMachineScaleSetProperties{
+				Overprovision: to.Ptr(false),
+				UpgradePolicy: &armcompute.UpgradePolicy{
+					Mode: to.Ptr(armcompute.UpgradeModeManual),
+				},
+				VirtualMachineProfile: &armcompute.VirtualMachineScaleSetVMProfile{
+					ExtensionProfile: &armcompute.VirtualMachineScaleSetExtensionProfile{
+						Extensions: []*armcompute.VirtualMachineScaleSetExtension{
+							{
+								Name: to.Ptr("vmssCSE"),
+								Properties: &armcompute.VirtualMachineScaleSetExtensionProperties{
+									Publisher:               to.Ptr("Microsoft.Azure.Extensions"),
+									Type:                    to.Ptr("CustomScript"),
+									TypeHandlerVersion:      to.Ptr("2.0"),
+									AutoUpgradeMinorVersion: to.Ptr(true),
+									Settings:                map[string]interface{}{},
+									ProtectedSettings: map[string]interface{}{
+										"commandToExecute": cseCmd,
+									},
+								},
+							},
+						},
+					},
+					OSProfile: &armcompute.VirtualMachineScaleSetOSProfile{
+						ComputerNamePrefix: to.Ptr(name),
+						AdminUsername:      to.Ptr("azureuser"),
+						CustomData:         &customData,
+						LinuxConfiguration: &armcompute.LinuxConfiguration{
+							SSH: &armcompute.SSHConfiguration{
+								PublicKeys: []*armcompute.SSHPublicKey{
+									{
+										KeyData: to.Ptr(string(pubKeyBytes)),
+										Path:    to.Ptr("/home/azureuser/.ssh/authorized_keys"),
+									},
+								},
+							},
+						},
+					},
+					StorageProfile: &armcompute.VirtualMachineScaleSetStorageProfile{
+						ImageReference: &armcompute.ImageReference{
+							ID: to.Ptr("/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/1804Gen2/versions/1.1677169694.31375"),
```

Comment:
nit: Should we move this (and similar long hardcoded strings) to variables? I can also move it in a different PR for testing e2e with latest VHD